helm/file_repo_store.go

```diff
@@ -0,0 +1,489 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/banzaicloud/pipeline/config"
+	phelm "github.com/banzaicloud/pipeline/pkg/helm"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	helmEnv "k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
+	"k8s.io/helm/pkg/repo"
+)
+
+type FileRepositoryStore struct {
+	Env helmEnv.EnvSettings
+}
+
+func NewFileRepositoryStore(orgName string) (*FileRepositoryStore, error) {
+	var helmPath = config.GetHelmPath(orgName)
+	env := createEnvSettings(fmt.Sprintf("%s/%s", helmPath, phelm.HelmPostFix))
+
+	repoStore := &FileRepositoryStore{Env: env}
+	err := repoStore.init()
+	if err != nil {
+		return nil, err
+	}
+	return repoStore, nil
+}
+
+// ReposGet returns repo
+func (s *FileRepositoryStore) ReposGet() ([]*repo.Entry, error) {
+
+	repoPath := s.Env.Home.RepositoryFile()
+	log.Debugf("Helm repo path: %s", repoPath)
+
+	f, err := repo.LoadRepositoriesFile(repoPath)
+	if err != nil {
+		return nil, err
+	}
+	if len(f.Repositories) == 0 {
+		return make([]*repo.Entry, 0), nil
+	}
+
+	return f.Repositories, nil
+}
+
+// ReposAdd adds repo(s)
+func (s *FileRepositoryStore) ReposAdd(Hrepo *repo.Entry) (bool, error) {
```

Comment:
```suggestion
func (s *FileRepositoryStore) ReposAdd(hrepo *repo.Entry) (bool, error) {
```

Please follow the variable naming convention.